orchestrator/core/llm/llm_agent_selector.py

```
@@ -848,51 +1051,352 @@ async def select_agents_for_subtasks(
+            workflow_ctx = default_ctx
+        
+        # BATCH PROCESSING - NO LOOPS!
+        logger.info(f"🚀 BATCH MODE: Processing ALL {len(subtasks)} subtasks at once")
+        
+        # Step 1: Query ALL agents ONCE
+        agents_result = await self._query_agents({"status": "available"})
+        all_agents = agents_result.get("matching_agents", [])
+        logger.info(f"  📊 Found {len(all_agents)} available agents")
+        
+        # Step 2: Batch semantic scoring for ALL subtasks
+        import asyncio
```

Ruff (0.14.5)

Redefinition of unused `asyncio` from line 18

Remove definition: `asyncio`

(F811)